src/libraries/Microsoft.PowerFx.Core/Public/Values/InMemoryRecordValue.cs

```diff
@@ -41,5 +41,34 @@ protected override bool TryGetField(FormulaType fieldType, string fieldName, out
         {
             return _fields.TryGetValue(fieldName, out result);
         }
+
+        public override FormulaValue UpdateFields(IEnumerable<KeyValuePair<string, FormulaValue>> record)
+        {
+            if (_fields.IsReadOnly)
+            {
+                return base.UpdateFields(record);
+            }
+
+            var fields = new List<NamedValue>();
+
+            foreach (var field in record)
+            {
+                var fieldName = field.Key;
+
+                // Throws if field is missing
+                Type.GetFieldType(fieldName);
+
+                if (field.Value is ErrorValue errorValue)
+                {
+                    return errorValue;
+                }
```

Comment:
is this right? 
Difference between:
- the whole record is an error 
- a field within the record is an error